plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -505,6 +563,17 @@ public static LocalDateTime fromPrestoTimestamp(long epochMicros)
         return LocalDateTime.ofInstant(instant, UTC);
     }
 
+    public static LocalDateTime fromLongTimestamp(LongTimestamp timestamp)
+    {
+        long epochSeconds = floorDiv(timestamp.getEpochMicros(), MICROSECONDS_PER_SECOND);
+        int microOfSecond = floorMod(timestamp.getEpochMicros(), MICROSECONDS_PER_SECOND);
+        int nanosOfSecond = (microOfSecond * NANOSECONDS_PER_MICROSECOND) +
+                floorDiv(timestamp.getPicosOfMicro(), PICOSECONDS_PER_NANOSECOND);
```

Comment:
`roundDiv`  looks appropriate, but would probably make this case to fail

CTAS `TIMESTAMP '..... .1234567499'`

this is because if we round up when going to picos to nanos, we get `.123456750'` and SQL Server will store 7-precision `.1234568`

However, if SQL Server did support precision up to 9, then `roundDiv` would be necessary here, not `floorDiv`

Thus, the actual code needs to take into account the limitations / capabilities of the remote database. 
To do this, we do not need a new parameter though. The caller has `timestampType`, which contains precision, so let's just pass desired precision into here.

The function can expect precision>6, so only the pico fraction needs to be rounded

- round `getPicosOfMicro()` to `12 - precision`
- scale to nanos

Reply:
Done! I found that similar rounding is done for time. I totally forgot that LDT is 9, not 12, thanks!